model-optimizer/mo/front/common/partial_infer/utils.py

```diff
@@ -109,4 +109,29 @@ def broadcast_shape(first_shape, second_shape):
         assert a_val == 1 or b_val == 1 or a_val == b_val, "Input shape do not broadcast"
         new_val = b_val if a_val == 1 else a_val
         new_shape[-i - 1] = new_val
-    return int64_array(new_shape)
\ No newline at end of file
+    return int64_array(new_shape)
+
+
+def get_shape_from_slice(input_shape: np.ndarray, slices: List[Union[int, slice]]) -> np.ndarray:
```

Comment:
I don't see unit tests for the updated version of the function

Reply:
Unittest for shape_infer function cover this function too.